swagger.yaml

```diff
@@ -5123,6 +5133,7 @@ definitions:
       - 'lso-requirements-satisfied'
       - 'ocs-requirements-satisfied'
       - 'cnv-requirements-satisfied'
+      - 'network-valid'
```

Comment:
I think that we should call it "network-type-valid"

Reply:
Why have a validation and not just return 400 upon update if it's invalid?

Reply:
the networkType will depend on the network (ipv6 vs 4) so there is the option it will not be define when the user updates the cluster params, no ?

Reply:
In hive, when we define the network-type we don't know if it will be valid for the current family (ipv4/ipv6).  Therefore, we need validation for it.

Reply:
We know it's valid if it doesn't return 400 right?

Reply:
Still, if it was defined during create cluster to be OpenshiftSDN, and later the discovered networks are IPv6, who will verify that we should not start installation this way ?

Reply:
So 400 will not be returned here

Reply:
so what about creating with a default (OVM) with no option for SDN?

Reply:
@ori-amizur If the discovered networks are IPv6 then we'll automatically choose IPv6 CIDR?

Reply:
What other choice we have? There are no IPv4 networks.

Reply:
I thought the user was choosing - I didn't realize the service was doing this automatically.
How does it work in [validations.ValidateVipDHCPAllocationWithIPv6](https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L2062)?

Reply:
Another point to keep in mind, Moran wants us as a default to use whatever OpenShift uses as a default in whatever version we're installing.  So we'll need to display to the user what type will be used.

Reply:
so @ori-amizur what do you think ? 
using a validation (and if yes can you maybe explain how does it work with  [validations.ValidateVipDHCPAllocationWithIPv6](https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L2062) or not giving an option to select a type when creating cluster?
or any alternative ?

Reply:
Maybe we can avoid the problem by checking the VIPs during cluster create.  If they are IPv6, then network-type must be OVN.
Same can be done during update cluster.

Reply:
Please rename to network-type-valid